pkg/sources/sources.go

```diff
@@ -50,6 +50,26 @@ type SourceUnitUnmarshaller interface {
 	UnmarshalSourceUnit(data []byte) (SourceUnit, error)
 }
 
+// SourceUnitEnumerator defines an optional interface a Source can implement to
+// support enumerating an initialized Source into SourceUnits.
+type SourceUnitEnumerator interface {
+	// Enumerate enumerates the initialized Source, outputting units. This
+	// method is synchronous but can be called in a goroutine to support
```

Comment:
Not necessarily disagreeing, but why make it synchronous instead of handling the concurrency inside `Enumerate`? Is the idea that the `units` chan will be shared for enumerating multiple sources?

Reply:
Mostly thinking that it's easier for a source to implement a synchronous function and have the `SourceManager` (when it exists) handle the concurrency.

For the initial implementation it'll be 1:1 unit channel to enumeration call. I don't anticipate it being shared across sources, since we need to know which source a unit is coming from.